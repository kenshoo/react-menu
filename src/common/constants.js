export const DIRECTIONS = {
  LEFT: "left",
  LEFT_TO_TOP: "left_to_top",
  RIGHT: "right",
  RIGHT_TO_TOP: "right_to_top",
  BOTTOM: "bottom",
  BOTTOM_LEFT: "bottom_left"
};

export default DIRECTIONS;
