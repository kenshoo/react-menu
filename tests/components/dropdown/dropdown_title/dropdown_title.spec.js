import React from "react";
import renderer from "react-test-renderer";

import { DIRECTIONS } from "../../../../src/common/constants";
import DropdownTitle from "../../../../src/components/dropdown/dropdown_title/dropdown_title";

describe("DropdownTitle component", () => {
  test("default snapshot", () => {
    const cmp = renderer.create(<DropdownTitle />);
    expect(cmp).toMatchSnapshot();
  });

  test("without icon", () => {
    const cmp = renderer.create(<DropdownTitle withIcon={false} />);
    expect(cmp).toMatchSnapshot();
  });

  test("renders children", () => {
    const cmp = renderer.create(<DropdownTitle>TEST</DropdownTitle>);
    expect(cmp).toMatchSnapshot();
  });

  test("className prop", () => {
    const cmp = renderer.create(
      <DropdownTitle className="TEST">TEST</DropdownTitle>
    );
    expect(cmp).toMatchSnapshot();
  });

  test("arrow direction - left", () => {
    const cmp = renderer.create(
      <DropdownTitle className="TEST" direction={DIRECTIONS.LEFT}>
        TEST
      </DropdownTitle>
    );
    expect(cmp).toMatchSnapshot();
  });

  test("arrow direction - bottom", () => {
    const cmp = renderer.create(
      <DropdownTitle className="TEST" direction={DIRECTIONS.BOTTOM}>
        TEST
      </DropdownTitle>
    );
    expect(cmp).toMatchSnapshot();
  });

  test("arrow direction - right", () => {
    const cmp = renderer.create(
      <DropdownTitle className="TEST" direction={DIRECTIONS.RIGHT}>
        TEST
      </DropdownTitle>
    );
    expect(cmp).toMatchSnapshot();
  });

  test("arrow direction - unknown", () => {
    const cmp = renderer.create(
      <DropdownTitle className="TEST" direction="TEST">
        TEST
      </DropdownTitle>
    );
    expect(cmp).toMatchSnapshot();
  });

  test("disabled prop", () => {
    const cmp = renderer.create(<DropdownTitle isDisabled>TEST</DropdownTitle>);
    expect(cmp).toMatchSnapshot();
  });

  test("disabledClassName prop", () => {
    const cmp = renderer.create(
      <DropdownTitle isDisabled disabledClassName="TEST">
        TEST
      </DropdownTitle>
    );
    expect(cmp).toMatchSnapshot();
  });
});
