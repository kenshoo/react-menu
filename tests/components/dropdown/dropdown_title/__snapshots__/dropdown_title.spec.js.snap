// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownTitle component arrow direction - bottom 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  TEST
  <svg
    className="sc-bdVaJa jBPbFr"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    />
  </svg>
</span>
`;

exports[`DropdownTitle component arrow direction - left 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  <svg
    className="sc-bdVaJa jBPbFr"
    height="20"
    viewBox="0 0 24 24"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14 8l-4 4 4 4"
      fill="none"
      fillRule="evenodd"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1"
    />
  </svg>
  TEST
</span>
`;

exports[`DropdownTitle component arrow direction - right 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  TEST
  <svg
    className="sc-bdVaJa jBPbFr"
    height="20"
    viewBox="0 0 24 24"
    width="20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 8l4 4-4 4"
      fill="none"
      fillRule="evenodd"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="1"
    />
  </svg>
</span>
`;

exports[`DropdownTitle component arrow direction - unknown 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  TEST
  <svg
    className="sc-bdVaJa jBPbFr"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    />
  </svg>
</span>
`;

exports[`DropdownTitle component className prop 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  TEST
  <svg
    className="sc-bdVaJa jBPbFr"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    />
  </svg>
</span>
`;

exports[`DropdownTitle component default snapshot 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  <svg
    className="sc-bdVaJa jBPbFr"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    />
  </svg>
</span>
`;

exports[`DropdownTitle component disabled prop 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  TEST
  <svg
    className="sc-bdVaJa jBPbFr"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    />
  </svg>
</span>
`;

exports[`DropdownTitle component disabledClassName prop 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  TEST
  <svg
    className="sc-bdVaJa jBPbFr TEST"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    />
  </svg>
</span>
`;

exports[`DropdownTitle component renders children 1`] = `
<span
  className="sc-bwzfXH cVmiUh"
>
  TEST
  <svg
    className="sc-bdVaJa jBPbFr"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    viewBox="0 0 448 512"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    />
  </svg>
</span>
`;
